### PR TITLE
[TDV-924] Assets vuex module

### PR DESCRIPTION
**Pull request by ardier16**

### CODE REVIEW

File: src/vuex/assets.e2e.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        state: {},
+        modules: { assets },
+      })
+    })
+
+    afterEach(() => {
+      sinon.restore()
+    })
+
+    it('factors', async () => {
```
Comment (max-nes) [inline src/vuex/assets.e2e.js]: what does `factors` mean to the test?

File: src/vuex/assets.module.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      assets.push(...pageResponse.data)
+    }
+
+    commit(vuexTypes.SET_ASSETS, assets)
+  },
+}
+
+export const getters = {
+  [vuexTypes.assets]: state => state.assets.map(a => new AssetRecord(a)),
+  [vuexTypes.getAssetByCode]: (_, getters) => assetCode =>
```
Comment (max-nes) [inline src/vuex/assets.module.js]: `assetByCode` is enough

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 Please check our [developers guide](https://gitlab.com/tokend/developers-guide)
 for further information about branching and tagging conventions.
 
+## [Unreleased]
+#### Added
+- Assets vuex module for loading and storing all the assets
+
+#### Updated
```
Comment (max-nes) [inline CHANGELOG.md]: Changed, not Updated

File: src/vuex/account.module.js [hunk 4]
```diff
@@ -33,6 +33,12 @@ export const actions = {
       include: ['balances.asset', 'balances.state'],
     })
 
+    commit(
+      vuexTypes.UPDATE_ASSETS,
+      account.balances.map(b => b.asset),
+      { root: true }
+    )
+
```
Comment (max-nes) [inline src/vuex/account.module.js]: Pls also add BalanceRecord. To keep backward compatibility, you can add a new getter for balances that will return you wrapped balances.

File: src/vuex/account.spec.js [hunk 5]
```diff
@@ -76,25 +76,23 @@ describe('account.module', () => {
...
-          [type]: payload,
-        }
-
-        await actions[vuexTypes.LOAD_ACCOUNT_BALANCES_DETAILS](store)
 
-        expect(store.commit.args)
-          .to
-          .deep
-          .equal(Object.entries(expectedMutations))
+        expect(store.commit.args).to.deep.equal([
```
Comment (max-nes) [inline src/vuex/account.spec.js]: pls use `have.been.calledOnceWithExactly()` instead

Comment (ardier16) [reply src/vuex/account.spec.js]: Used

File: src/vuex/account.module.js (hunk 4 above)
Comment (ardier16) [reply src/vuex/account.module.js]: Added

File: CHANGELOG.md (hunk 3 above)
Comment (ardier16) [reply CHANGELOG.md]: Renamed

File: src/vuex/assets.module.js (hunk 2 above)
Comment (ardier16) [reply src/vuex/assets.module.js]: Ok, renamed

File: src/vuex/assets.e2e.js (hunk 1 above)
Comment (ardier16) [reply src/vuex/assets.e2e.js]: Fixed, sorry

File: src/js/records/entities/balance.record.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+export class BalanceRecord {
+  constructor (record = {}) {
+    this._record = record
+
+    this.id = record.id
+    this.balance = safeGet(record, 'state.available')
+    this.convertedBalance = record.convertedBalance
+
+    this.assetDetails = new AssetRecord(record.asset)
+    this.asset = this.assetDetails.code
```
Comment (max-nes) [inline src/js/records/entities/balance.record.js]:
```js
    this.asset = new AssetRecord(record.asset)
```

is enough, asset code can be easily derived from .asset.code

File: src/vuex/assets.module.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  async [vuexTypes.LOAD_ASSETS] ({ commit }) {
+    let pageResponse
+    let assets
+
+    pageResponse = await api.get('/v3/assets', {
+      page: { limit: ASSETS_PAGE_LIMIT },
+    })
+    assets = pageResponse.data
+
+    while (pageResponse.data.length) {
```
Comment (max-nes) [inline src/vuex/assets.module.js]:
```suggestion
    while (pageResponse.data.length === ASSETS_PAGE_LIMIT) {
```

File: src/js/records/entities/balance.record.js (hunk 6 above)
Comment (ardier16) [reply src/js/records/entities/balance.record.js]: Updated

**Pull request merged**